mlflow/langchain/_langchain_autolog.py

```diff
@@ -180,6 +228,682 @@ def _chain_with_retriever(model):
     return False
 
 
+def get_mlflow_langchain_callback():
+    from langchain.callbacks.base import BaseCallbackHandler
+    from langchain_core.agents import AgentAction, AgentFinish
+    from langchain_core.documents import Document
+    from langchain_core.load import dumpd
+    from langchain_core.messages import BaseMessage
+    from langchain_core.outputs import (
+        ChatGeneration,
+        ChatGenerationChunk,
+        GenerationChunk,
+        LLMResult,
+    )
+    from tenacity import RetryCallState
+
+    from mlflow.entities import SpanEvent, SpanType
+    from mlflow.entities.span_status import SpanStatus
+    from mlflow.entities.trace import Trace
+    from mlflow.entities.trace_status import TraceStatus
+    from mlflow.tracing.types.wrapper import MLflowSpanWrapper
+
+    class MLflowLangchainCallback(BaseCallbackHandler, metaclass=ExceptionSafeAbstractClass):
+        """
+        Callback for auto-logging artifacts.
+        We need to inherit ExceptionSafeAbstractClass to avoid invalid new
+        input arguments added to original function call.
+        Args:
+            tracking_uri: MLflow tracking server uri.
+            experiment_name: Name of the experiment.
+            run_id: Id of the run to log the artifacts.
+        """
+
+        def __init__(
+            self,
+            experiment_name="langchain_autologging",
+            run_id=None,
+            *,
+            tags: Optional[Dict[str, Any]] = None,
+            artifacts_dir: str = "",
+        ):
+            super().__init__()
+            self._mlflow_client = MlflowClient()
+            # create experiment if not exists
```

Comment:
Could we move this logic to tracing APIs? I left it as TODO [here](https://github.com/mlflow/mlflow/blob/tracing/mlflow/tracing/trace_manager.py#L128) but this solution seems to be applicable, so would be great if we can use this to general cases:)

Reply:
Got it. IIRC the experiment_id is mandatory field for trace so we need to create one if not exist. But it's not the scope of this PR, I will follow-up to confirm the requirement.